deps/rabbit/src/rabbit_quorum_queue.erl

```diff
@@ -442,17 +442,25 @@ become_leader0(QName, Name) ->
 all_replica_states() ->
     Rows0 = ets:tab2list(ra_state),
     Rows = lists:map(fun
-                         ({K, follower, promotable}) ->
-                             {K, promotable};
-                         ({K, follower, non_voter}) ->
-                             {K, non_voter};
-                         ({K, S, _}) ->
-                             %% voter or unknown
-                             {K, S};
-                         (T) ->
-                             T
+                        (T = {K, _, _}) ->
+                            case rabbit_process:is_registered_process_alive(K) of
+                                true ->
+                                    to_replica_state(T);
+                                false ->
+                                    []
```

Comment:
I'm not sure about returning nil here. is that expected?

Reply:
hmm, I think the `noproc` issue, if my understanding this right, is when we have an entry in `ra_state` of a member which had already terminated, but wasnt cleaned up from ets from FSM terminate callback. Then its still listed here as if it were still active, yet not alive. But if alive=false, we should handle it as if it had been removed from `ra_state`, so returning nil here. (i'm not sure if we should also be doing a deletion from `ra_state` here when the fsm proc is not alive)

Reply:
all other code paths return a tuple `{RegName, Status}` so returning `[] feels a bit odd, could you either return `{K, noproc}` or use lists:filtermap to remove the entry all together?

Reply:
Ok,  `lists:filtermap` sounds good. this was removing/flattening out the `[]` on the conversion to a `map`, but no need for that extra step. updated 👍